contrib/grimshot

```diff
@@ -0,0 +1,101 @@
+#!/bin/sh
+
+## Grimshot: a helper for screenshots within sway
+## Requirements:
+##  - `grim`: screenshot utility for wayland
+##  - `slurp`: to select an area
+##  - `swaymsg`: to read properties of current window
+##  - `wl-copy`: clipboard utility
+##  - `jq`: json uliity to parse swaymsg output
+##  - `notify-send`: to show notifications
+## Those are needed to be installed, if unsure, run `grimshot check`
+##
+## Examples:
+## `grimshot copy win` - to copy current window
+## `grimshot save area` - to select area and save it to default file (Pictures/Grimshot-$datetime.png)
+## `grimshot save screen ~/screenshot.png` - to save screenshot under ~/screenshot.png
+## `grimshot` - usage
+## `grimshot check` - verify if tools are installed
+
+ACTION=${1:-usage}
+SUBJECT=${2:-screen}
+FILE=${3:-$(xdg-user-dir PICTURES)/$(date +'Grimshot %Y-%m-%d %H-%M-%S.png')}
+if [ $ACTION = "usage" ] ; then
+  echo "Usage:"
+  echo "  grimshot copy|save win|screen|area [FILE]"
+  echo "Troubleshoot:"
+  echo "  grimshot check"
+  exit
+fi
+function notify() {
+  CMD="notify-send -t 3000 -a grimshot $*"
+  sh -c "$CMD"
+}
+
+function notifyOk() {
+  TITLE=${2:-"Screenshot"}
+  MESSAGE=${1:-"OK"}
+  notify "\"$TITLE\"" "\"$MESSAGE\""
+}
+function notifyError() {
+  TITLE=${2:-"Screenshot"}
+  MESSAGE=${1:-"Error taking screenshot with grim"}
+  notify -u critical "\"$TITLE\"" "\"$MESSAGE\""
+}
+
+function die() {
+  MSG=${1:-Bye}
+  echo "Dead: $MSG"
+  notify-send -u critical -a grimshot "Screenshot" "Dead: $MSG"
+  exit -1
+}
+function check() {
+  COMMAND=$1
+  command -v $COMMAND > /dev/null 2>&1
+  [[ $? = 0 ]] && RESULT="OK" || RESULT="NOT FOUND"
+  echo "   $COMMAND: $RESULT"
+}
+
+if [ $ACTION = "check" ] ; then
+  echo "Checking if required tools are installed. If something is missing, install it to your system and make it available in PATH..."
+  check grim
+  check slurp
+  check swaymsg
+  check wl-copy
+  check jq
+  check notify-send
+  exit
+elif [ $SUBJECT = "area" ] ; then
+  GEOM=$(slurp -d)
+  GEOM="-g \"$GEOM\""
+  WHAT="Area"
+elif [ $SUBJECT = "win" ] ; then
+  GEOM=`swaymsg -t get_tree | jq -r '.. | (.nodes? // empty)[] | select(.focused) | .rect | "\(.x),\(.y) \(.width)x\(.height)"'`
+  GEOM="-g \"$GEOM\""
+  APP_ID=`swaymsg -t get_tree | jq -r '.. | (.nodes? // empty)[] | select(.focused) | .app_id'`
+  WHAT="$APP_ID window"
+elif [ $SUBJECT = "screen" ] ; then
+  GEOM=""
+  WHAT="Screen"
+else
+  die "Unknown subject to take a screen shot from" $SUBJECT
+fi
+
+if [ $ACTION = "copy" ] ; then
+  TMP=${TMPDIR:-/tmp}/grim-$USER-data.png
+  CMD="grim $GEOM $TMP"
```

Comment:
The current way of invoking external tools looks very fragile.
I think you should either use arrays if you decide to switch to `bash` instead of `sh`, or use something like this instead:
```
set -- 'mycommand' 'needs some complex' 'args'
"$@"
```
with arrays it would be for example:
```
GEOM=("-g" "$GEOM")
grim "${GEOM[@]}" "$FILE"
```

Reply:
Working on this one...